src/nft_contract/src/main.nr

```diff
@@ -0,0 +1,564 @@
+
+    /// @return name The name of the NFT collection
+    #[public]
+    #[view]
+    fn public_get_name() -> pub FieldCompressedString {
+        storage.name.read()
+    }
+
+    /// @return symbol The symbol of the NFT collection
+    #[public]
+    #[view]
+    fn public_get_symbol() -> pub FieldCompressedString {
+        storage.symbol.read()
+    }
+
+    /// @return owner The owner of a token by id
+    #[public]
+    #[view]
+    fn public_owner_of(token_id: Field) -> pub AztecAddress {
+        storage.public_owners.at(token_id).read()
+    }
+
+    /** ==========================================================
+     * ===================== UNCONSTRAINED =======================
+     * ======================================================== */
+
+    /// @notice Returns an array of token IDs owned by `owner` in private and a flag indicating whether a page limit was reached
+    /// @dev Starts getting the notes from page with index `page_index`
+    /// @dev Zero values in the array are placeholder values for non-existing notes
+    /// @param owner The address of the owner
+    /// @param page_index The index of the page to start getting notes from
+    /// @return owned_nft_ids An array of token IDs owned by `owner`
+    /// @return page_limit_reached A flag indicating whether a page limit was reached
+    #[utility]
+    pub unconstrained fn get_private_nfts(
+        owner: AztecAddress,
+        page_index: u32,
+    ) -> pub ([Field; MAX_NOTES_PER_PAGE], bool) {
+        let offset = page_index * MAX_NOTES_PER_PAGE;
+        let mut options = NoteViewerOptions::new();
+        let nfts = storage.private_nfts.at(owner).view_notes(options.set_offset(offset));
+
+        let mut owned_nft_ids = [0; MAX_NOTES_PER_PAGE];
+        for i in 0..options.limit {
+            if i < nfts.len() {
+                owned_nft_ids[i] = nfts.get_unchecked(i).token_id;
+            }
+        }
+        let page_limit_reached = nfts.len() == options.limit;
+        (owned_nft_ids, page_limit_reached)
+    }
+
+    /** ==========================================================
+    * ======================= MINTABLE ==========================
+    * ======================================================== */
+
+    /// @notice Mints a token to a private address
+    /// @dev Asserts that the caller is an authorized minter
+    /// @param to The address of the recipient
+    /// @param token_id The id of the token to mint
+    #[private]
+    fn mint_to_private(to: AztecAddress, token_id: Field) {
+        assert(token_id != 0, "zero token ID not supported");
+        _validate_minter(context.msg_sender(), storage.minter.read());
+
+        NFT::at(context.this_address()).set_nft_exists_internal(token_id, true).enqueue(&mut context);
+        storage.private_nfts.at(to).insert(NFTNote::new(token_id, to)).emit(encode_and_encrypt_note(
+            &mut context,
+            to,
+            to,
+        ));
+    }
```

Comment:
i think we do need an arbitrary `from` here no? just for encrypting the note
also, perhaps `from` isn't the happiest of names (both here and for token)